cocotb/decorators.py

```diff
@@ -214,11 +221,23 @@ def __await__(self):
 
     __bool__ = __nonzero__
 
+    @property
+    def funcname(self):
+        warnings.warn("This was for internal use only and will be removed",
```

Comment:
```suggestion
        warnings.warn("`RunningCoroutine.funcname` was for internal use only and will be removed",
```
Same for the others.